org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTConverter.java

```diff
@@ -3630,15 +3630,35 @@ public TypeParameter convert(org.eclipse.jdt.internal.compiler.ast.TypeParameter
 		return typeParameter2;
 	}
 
+	@SuppressWarnings("cast")
 	public Pattern convert(org.eclipse.jdt.internal.compiler.ast.TypePattern pattern) {
-		TypePattern typePattern = new TypePattern(this.ast);
+		TypePattern typePattern;
+		if(this.ast.apiLevel < AST.JLS22_INTERNAL) {
+			typePattern = new TypePattern(this.ast);
+		} else {
+			if(pattern.local != null && !(String.valueOf(pattern.local.name[0]) == "_") && pattern.local.type != null) { //$NON-NLS-1$
+				typePattern = new TypePattern(this.ast, true);//SingleVariableDeclaration
+			} else {
+				typePattern = new TypePattern(this.ast, false);//VariableDeclarationFragment
+			}
+		}
+
 		if (this.resolveBindings) {
 			recordNodes(typePattern, pattern);
 		}
 		if (pattern.local == null) {
 			return createFakeNullPattern(pattern);
 		}
-		typePattern.setPatternVariable(convertToSingleVariableDeclaration(pattern.local));
+
+		if(this.ast.apiLevel < AST.JLS22_INTERNAL) {
+			typePattern.setPatternVariable(convertToSingleVariableDeclaration(pattern.local));
+		} else {
+			if(pattern.local != null && !(String.valueOf(pattern.local.name[0]) == "_") && pattern.local.type != null) { //$NON-NLS-1$
+				typePattern.setPatternVariable((VariableDeclaration)convertToSingleVariableDeclaration(pattern.local));
+			} else {
+				typePattern.setPatternVariable((VariableDeclaration) convertToVariableDeclarationFragment(pattern.local));
```

Comment:
can there be a case where pattern.local == null. In such a case what happens in convToVarDeclFragmment(null) - please check

Reply:
Done. I have changed the position where the createFakeNullPattern method is called